web-ng/src/app/components/form-field-editor/form-field-editor.component.ts

```diff
@@ -112,6 +126,55 @@ export class FormFieldEditorComponent implements OnInit, OnChanges {
    */
 
   onFieldTypeSelect(event: FieldTypeSelectOption) {
+    this.type = event.type.toString();
     this.formFieldGroup.patchValue({ type: event });
+    if (event.type === 2) {
+      this.onAddOption();
+    } else {
+      this.formOptions = undefined;
+    }
+  }
+
+  /**
+   * Emits event to layer dialog whenvever label or code gets updated in options-editor.
+   *
+   * @param event: label and code value of the option field.
+   * @param index: index of the option to be updated.
+   * @returns void
+   *
+   */
+
+  onOptionUpdate(event: { label: string; code: string }, index: number) {
+    const optionId = this.dataStoreService.generateId();
+    const option = new Option(
+      optionId || '',
+      event.code,
+      StringMap({ en: event.label })
+    );
+    this.setOption(index, option);
+  }
+
+  trackByFn(index: number) {
+    return index;
+  }
+
+  onAddOption() {
+    const optionId = this.dataStoreService.generateId();
+    const option = new Option(optionId || '', '', StringMap({ en: '' }));
+    const index = this.formOptions?.options.size || 0;
+    this.setOption(index, option);
+  }
+
+  setOption(index: number, option: Option) {
+    const cardinality = this.formOptions?.cardinality || 2;
+    let options = this.formOptions?.options || List<Option>();
+    options = options?.set(index, option);
+    this.multipleChoice = new MultipleChoice(cardinality, options);
+    this.update.emit({
```

Comment:
It looks like the form editor and option component keep their state in sync on each change. Is this necessary? Can we better encapsulate the behavior of the options component so that the parent component can query its state as needed?

Reply:
I actually may be wrong here - how to passive components in Angular normally expose their state? Do parents subscribe to  `output`, or do the invoke a method to query state?

Reply:
It works in both way you can use value change to send value to the parent and on the other way you can query child state too. Both ways are valid.